packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -280,6 +280,10 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
     islandContext
   }
 
+  if (import.meta.prerender && process.env.NUXT_SHARED_DATA) {
+    ssrContext._sharedStorage = useStorage('internal:nuxt:prerender:shared')
```

Comment:
Herer, we could expose a wrapped cached function for the handler, which can handle concurrency as well. Supporting concurrency over cache I think could make this feature significantly more useful!

Reply:
Nice idea!

My only worry was memory consumption but I guess we could use an lru cache? Wdyt?

Reply:
PS: Using `.cache/nitro` as key, we could benefit from single configuration to define where build cache goes.

Reply:
> My only worry was memory consumption but I guess we could use an lru cache? Wdyt? 

Surely! We could also mount an LRU for a specific subpath and configure it for nuxt pretender specifically. I am thinking also for main `.cache` we could do the same later.

Reply:
Also, would we ever prerender in a different context, meaning we needed storage to share data?

Reply:
At some point yes, but it will also come with an internal API that we can integrate with the storage layer so we won't "loose" this feature since we rely on storage API.

Reply:
implemented wrapped function handling with storage layer underneath. ✅ 

the more I think about it, the more I think we should have a hybrid memory/FS storage for all of these caches - I can implement in separate PR. The idea would be that we have an LRU cache backed up by a FS cache so we _never_ have to rerun a fetcher function. The reason we need FS backing is that the other caches (payload particularly) _must_ retain the data - there is no fallback for them. And on sufficiently large sites the payloads could end up taking gigabytes of memory.

Reply:
Using FS as the main cache + hybrid memory/LRU (or TTLed) makes sense for this usage. 

It could be nice also to see where we can have the cache so as not to reinvent the same thing in multiple places. Nitro caching API also handles concurrent promises with a storage-backed cache and TTL support (it will be a standalone lib) unstorage/nitro interaction should support multi-layer caching we could collaborate on these efforts.